docs/content/manual/pre-release/ha/ha-volume-migration.md

```diff
@@ -2,30 +2,400 @@
+```
+
+- "Waiting to confirm migration until migration engine is ready"
+- "Confirming migration"
+- No detachment
+- New engine and replicas
+
+#### 1.2. Confirmation immediately before crash
+
+Confirmation succeeds. Then, the volume detaches from and reattaches to the new node.
+
+```bash
+k delete --wait=false -f va-1.yaml && sleep 0.5 && kl delete pod instance-manager-ea5f8778d6c99e747289ff09c322d75a
+# OR
+k delete --wait=false -f va-1.yaml && sleep 1 && kl delete pod instance-manager-ea5f8778d6c99e747289ff09c322d75a
+```
+
+- "Confirming migration"
+- "...selected to detach from <new>"
+- "...selected to attach to <new>"
+- New engine and replicas
+
+#### 1.3. Rollback immediately before or after crash
+
+Rollback succeeds.
+
+```bash
+kl delete --wait=false pod instance-manager-ea5f8778d6c99e747289ff09c322d75a && sleep 0.5 && k delete -f va-2.yaml
+# OR
+kl delete --wait=false pod instance-manager-ea5f8778d6c99e747289ff09c322d75a && sleep 1 && k delete -f va-2.yaml
+# OR
+k delete --wait=false -f va-2.yaml && sleep 0.5 && kl delete pod instance-manager-ea5f8778d6c99e747289ff09c322d75a
+# OR
+k delete --wait=false -f va-2.yaml && sleep 1 && kl delete pod instance-manager-ea5f8778d6c99e747289ff09c322d75a
+```
+
+- "Rolling back migration"
+- No detachment
+- Same engine and replicas
+
+### 2. Old engine crash
+
+Crash the engine on the old node by killing its instance-manager pod.
+
+#### 2.1 No immediate confirmation or rollback
+
+The volume completely detaches and remains detached. Logs indicate next steps. Deleting either of the two
+VolumeAttachments gets the volume unstuck.
+
+```bash
+kl delete pod instance-manager-699da83c0e9d22726e667344227e096b
+```
+
+- "...selected to detach from <old>"
+- "Cancelling migration for detached volume..."
+- MigrationFailed event
+- "Volume migration between <old> and <new> failed; detach volume from extra node to resume"
+
+```
+kubectl -n longhorn-system get volume -oyaml | grep -i nodeid
+    migrationNodeID: ""
+    nodeID: ""
+    currentMigrationNodeID: ""
+    currentNodeID: ""
+    pendingNodeID: ""
+```
+
+```bash
+kl delete -f va-2.yaml
+```
+
+```
+kubectl -n longhorn-system get volume -oyaml | grep -i nodeid
+    migrationNodeID: ""
+    nodeID: eweber-v126-worker-9c1451b4-kgxdq
+    currentMigrationNodeID: ""
+    currentNodeID: eweber-v126-worker-9c1451b4-kgxdq
+    pendingNodeID: ""
+```
+
+#### 2.2 Confirmation immediately after crash
+
+The volume automatically detaches from the old node. Then, it reattaches to the new node.
+
+```bash
+kl delete --wait=false pod instance-manager-699da83c0e9d22726e667344227e096b && sleep 0.5 && k delete -f va-1.yaml
+# OR
+kl delete --wait=false pod instance-manager-699da83c0e9d22726e667344227e096b && sleep 1 && k delete -f va-1.yaml
+```
+
+- "...selected to detach from <old>"
+- "Cancelling migration for detached volume..."
+- MigrationFailed event
+- "...selected to attach to <new>"
+- Same engine and replicas
```

Comment:
Since the volume has been attached to the new node, what does "same engine and replicas" mean here?

Reply:
Since the migration actually failed, the new engine and replicas created for the migration should be cleaned up. Then, the old engine and replicas are used to attach to the new node. It's a migration failure -> detach -> attach instead of a successful migration.

In practice, it means if you check the engine and replica names before the scenario and then after, they are the same. If the migration succeeded, they would be different.